.github/scripts/promote_branch.sh

```diff
@@ -0,0 +1,163 @@
+# it performs a git force push. There is no pull request.
+#
+# Checks:
+#   - If there is content in the staging branch that is not yet in the production branch, the
+#     script will not git push to add more content to the staging branch. This can be overridden with
+#     --force-to-staging true
+#   - If promoting to production and the content has not been in the staging branch for at least 7 days,
+#     the script will exit without doing a push. Content is expected to sit in staging for at least a week
+#     to provide sufficient testing time. This can be overridden with --override true
+#
+# Prerequisities:
+#   - curl, git and jq installed.
+
+set -e
+
+# GitHub repository details
+ORG="konflux-ci"
+REPO="community-catalog"
+
+OPTIONS=$(getopt --long "promotion-type:,force-to-staging:,override:,dry-run:,help" -o "p:,h" -- "$@")
+eval set -- "$OPTIONS"
+while true; do
+    case "$1" in
+        -p|--promotion-type)
+            PROMOTION_TYPE="$2"
+            shift 2
+            ;;
+        --force-to-staging)
+            FORCE_TO_STAGING="$2"
+            shift 2
+            ;;
+        --override)
+            OVERRIDE="$2"
+            shift 2
+            ;;
+        --dry-run)
+            DRY_RUN="$2"
+            shift 2
+            ;;
+        -h|--help)
+            print_help
+            exit
+            ;;
+        --)
+            shift
+            break
+            ;;
+        *) echo "Error: Unexpected option: $1" >&2
+    esac
+done
+
+cleanup() {
+  if [ -d "${1}" ]; then
+    echo "Deleting tmpDir..."
+    cd -
+    rm -rf "${1:?}"
+  fi
+}
+
+print_help() {
+    echo "Usage: $0 --promotion-type branch1-to-branch2 [--force-to-staging false] [--override false] [--dry-run false]"
+    echo
+    echo "  --promotion-type:   The type of promotion to perform. Either development-to-staging"
+    echo "                      or staging-to-production."
+    echo "  --force-to-staging: If passed with value true, allow promotion to staging even"
+    echo "                      if staging and production differ."
+    echo "  --override:         If passed with value true, allow promotion to production"
+    echo "                      even if the change has not been in staging for one week."
+    echo "  --dry-run:          If passed with value true, print out the changes that would"
+    echo "                      be promoted but do not git push or delete the temp repo."
+    echo
+    echo "  --promotion-type has to be specified."
+}
+
+check_if_branch_differs() {
+    ACTUAL_DIFFERENT_LINES=$(git diff --numstat "origin/$1" | wc -l)
+    if [ "$ACTUAL_DIFFERENT_LINES" -ne 0 ] ; then
+        echo "Lines differ in branch $1"
+        echo "Actual differing lines: $(git diff --numstat origin/"$1")"
+        exit 1
+    fi
+}
+
+check_if_any_commits_in_last_week() {
+    NEW_COMMITS=$(git log --oneline --since="$(date --date="6 days ago" +%Y-%m-%d)" | wc -l)
```

Comment:
So it is effectively 6 days and not a week, right? When I read the description I was worried that if it was really 7 days, it might be a potential problem if we run it every week. Maybe it would make sense to say "6 days" instead of "a week" everywhere?